spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheProperties.java

```diff
@@ -233,4 +242,72 @@ public void setConfig(Resource config) {
 
 	}
 
+	/**
+	 * Redis-specific cache properties. Properties set will be used as the defaults for
+	 * all Redis caches.
+	 *
+	 * @see org.springframework.data.redis.cache.RedisCacheConfiguration
+	 */
+	public static class Redis {
+
+		/**
+		 * The TTL for entries. By default, entries do not expire, a value of zero
+		 * indicates an eternal entry, and the given value will be converted to seconds..
+		 *
+		 * @see Duration#parse(CharSequence)
+		 */
+		private Duration ttl = Duration.ZERO;
+
+		/**
+		 * Whether to allow caching of {@literal null} values.
+		 */
+		private boolean cacheNullValues = true;
+
+		/**
+		 * The non-null Redis key prefix to use instead of the default one.
+		 */
+		private String keyPrefix;
+
+		/**
+		 * Whether to use the key prefix when writing to Redis.
+		 */
+		private boolean useKeyPrefix = true;
+
+		public Duration getTtl() {
+			return this.ttl;
+		}
+
+		public Redis setTtl(Duration ttl) {
+			this.ttl = ttl;
+			return this;
+		}
+
+		public boolean isCacheNullValues() {
+			return this.cacheNullValues;
+		}
+
+		public Redis setCacheNullValues(boolean cacheNullValues) {
+			this.cacheNullValues = cacheNullValues;
+			return this;
+		}
+
+		public Optional<String> getKeyPrefix() {
```

Comment:
Configuration properties must be JavaBean properties (the type must match) and we don't support `Optional` here.